### PR TITLE
[FLINK-22155][table] EXPLAIN statement should validate insert and query separately

### PR DESCRIPTION
**Pull request by fsk119**
## What is the purpose of the change

*Validate insert and query separately in EXPLAIN.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java [hunk 1]
```diff
@@ -844,16 +843,19 @@ private Operation convertShowViews(SqlShowViews sqlShowViews) {
...
-                || sqlExplain.getFormat() != SqlExplainFormat.TEXT) {
-            throw new TableException("Only default behavior is supported now, EXPLAIN PLAN FOR xx");
+    /** Convert RICH EXPLAIN statement. */
+    private Operation convertRichExplain(SqlRichExplain sqlExplain) {
+        Operation operation;
+        SqlNode sqlNode = sqlExplain.getStatement();
+        if (sqlNode instanceof RichSqlInsert) {
+            operation = convertSqlInsert((RichSqlInsert) sqlNode);
+        } else if (sqlNode instanceof SqlSelect) {
+            operation = convertSqlQuery(sqlExplain.getStatement());
```
Comment (cuspymd) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java:853]: Can't `sqlNode` be passed as a parameter?

Comment (fsk119) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java:853]: Flink has its own logic to validate the INSERT statement, which is different from calcite. Here we only validate the query part of the INSERT statement and check whether the sink schema is as same as the query schema if the statement is the INSERT statement.

Comment (cuspymd) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java:853]: Do you mean that the results of executions 1 and 2 below are different?
```
// 1
convertSqlQuery(sqlExplain.getStatement());
// 2
convertSqlQuery(sqlNode);
```

Comment (fsk119) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java:853]: I mean using `SqlValidatorImpl#validate(insertStatement)` may have the different plans comparing to using `SqlToOperationConverter#convertSqlInsert`. The validation about query is the same...

**Pull request closed without merging**